misk/src/main/kotlin/misk/web/exceptions/GrpcExceptionMapper.kt

```diff
@@ -0,0 +1,44 @@
+package misk.web.exceptions
+
+import com.squareup.wire.GrpcException
+import com.squareup.wire.GrpcStatus
+import misk.proto.Status
+import misk.web.Response
+import misk.web.ResponseBody
+import misk.web.mediatype.MediaTypes
+import misk.web.toResponseBody
+import okhttp3.Headers
+import org.slf4j.event.Level
+import wisp.logging.getLogger
+import java.io.IOException
+import java.net.HttpURLConnection
+import javax.inject.Inject
+
+internal class GrpcExceptionMapper @Inject internal constructor() : ExceptionMapper<GrpcException> {
+  private val logger = getLogger<GrpcExceptionMapper>()
+
+  override fun toResponse(th: GrpcException) = INTERNAL_SERVER_ERROR_RESPONSE
+
+  override fun toGrpcResponse(th: GrpcException): GrpcErrorResponse {
+    val status = try {
+      th.grpcStatusDetails?.let { Status.ADAPTER.decode(it) }
+    } catch (e: IOException) {
+      logger.error(e) {
+        "Received invalid grpc-status-details-bin header: ${th.grpcStatusDetails}"
+      }
+      null
+    }
+
+    return GrpcErrorResponse(th.grpcStatus, th.grpcMessage, status?.details ?: listOf())
```

Comment:
I don't think we want to do this as the default in Misk. This may leak internal details to other services. I'd probably just remove the `toGrpcResponse` method here and have it use the default one that's in `ExceptionHandlingInterceptor`.

Reply:
You're right.